src/test/java/teammates/test/cases/webapi/UpdateInstructorPrivilegeActionTest.java

```diff
@@ -0,0 +1,309 @@
+        reqBody.setCanSubmitSessionInSections(true);
+        reqBody.setFeedbackSessionName("session1");
+
+        assertThrows(InvalidHttpRequestBodyException.class, () -> {
+            UpdateInstructorPrivilegeAction illegalAction = getAction(reqBody, submissionParams);
+            getJsonResult(illegalAction);
+        });
+    }
+
+    @Test
+    protected void testExecute_withCourseIdAndSectionNameAndCourseLevelPrivilege_shouldFail() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.INSTRUCTOR_ID, instructor1OfCourse1.getGoogleId(),
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId,
+        };
+
+        InstructorPrivilegeUpdateRequest reqBody = new InstructorPrivilegeUpdateRequest();
+        reqBody.setCanModifyInstructor(true);
+        reqBody.setSectionName("section1");
+
+        assertThrows(InvalidHttpRequestBodyException.class, () -> {
+            UpdateInstructorPrivilegeAction illegalAction = getAction(reqBody, submissionParams);
+            getJsonResult(illegalAction);
+        });
+    }
+
+    @Test
+    protected void testExecute_withCourseIdAndSectionNameAndSessionNameAndSectionLevelPrivilege_shouldFail() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.INSTRUCTOR_ID, instructor1OfCourse1.getGoogleId(),
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId,
+        };
+
+        InstructorPrivilegeUpdateRequest reqBody = new InstructorPrivilegeUpdateRequest();
+        reqBody.setCanViewStudentInSections(true);
+        reqBody.setSectionName("section1");
+        reqBody.setFeedbackSessionName("session1");
+
+        assertThrows(InvalidHttpRequestBodyException.class, () -> {
+            UpdateInstructorPrivilegeAction illegalAction = getAction(reqBody, submissionParams);
+            getJsonResult(illegalAction);
+        });
+    }
+
+    @Test
+    protected void testExecute_noPrivilegesInRequestBody_shouldFail() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.INSTRUCTOR_ID, instructor1OfCourse1.getGoogleId(),
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId,
+        };
+
+        InstructorPrivilegeUpdateRequest reqBody = new InstructorPrivilegeUpdateRequest();
+
+        assertThrows(InvalidHttpRequestBodyException.class, () -> {
+            UpdateInstructorPrivilegeAction illegalAction = getAction(reqBody, submissionParams);
+            getJsonResult(illegalAction);
+        });
+    }
+
+    @Test
+    protected void testExecute_withInvalidInstructorId_shouldFail() {
```

Comment:
You should have one for invalid course id?

Reply:
invalid course id is in the scope of access control

Reply:
Then you should add a check for invalid course id in access control?